src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int baseA;
+    private int baseB;
+    private int leg;
+    private double height;
+
+    public IsoscelesTrapezoid() {
+        super();
+        this.baseA = new Random().nextInt(100);
+        this.baseB = new Random().nextInt(100);
+        this.leg = new Random().nextInt(100);
+        this.height = Math.sqrt(Math.pow(leg,2) - Math.pow((baseA - baseB),2));
```

Comment:
The calculation of height assumes that the trapezoid is isosceles and uses the Pythagorean theorem. Ensure that `baseA` is greater than `baseB` for this calculation to be valid, or adjust the logic accordingly.